tests/test_resources/test_modules/test_functions/test_function.py

```diff
@@ -173,301 +75,469 @@ async def async_slow_generator(size):
+        res = remote_summer.get_or_call(f"my_run_{time.time_ns()}", False, True, 12, 5)
+        assert res == 17
+
+        # not first run - implement later (raises an exception).
+        remote_run = remote_summer.run(2, 3)
+        get_or_call_res = remote_summer.get_or_call(
+            run_name=remote_run
+        ).resolved_state()
+        # TODO: should not use .resolved_state(), need to be fixed.
+        assert get_or_call_res == 5
+
+    @pytest.mark.skip(
+        "keep warm can not accept autostop_min arg, since cluster.keep_warm do not accept this arg."
     )
-    sum1 = requests.post(
-        "http://127.0.0.1:80/call/remote_function/call",
-        json={"args": [1, 2]},
-        auth=auth,
-    ).json()
-    assert int(sum1) == 3
-    sum2 = requests.post(
-        "http://127.0.0.1:80/call/remote_function/call",
-        json={"kwargs": {"a": 1, "b": 2}},
-        auth=auth,
-    ).json()
-    assert int(sum2) == 3
-
-    tun.close()
-
-
-@unittest.skip("Not yet implemented.")
-def test_http_url_with_curl():
-    # NOTE: Assumes the Function has already been created and deployed to running cluster
-    s = rh.function(name="test_function")
-    curl_cmd = s.http_url(a=1, b=2, curl_command=True)
-    print(curl_cmd)
-
-    # delete_configs the function data from the RNS
-    delete_function_from_rns(s)
-
-    assert True
-
-
-# test that deprecated arguments are still backwards compatible for now
-@pytest.mark.clustertest
-def test_reqs_backwards_compatible(cluster):
-    summer_cpu = rh.function(fn=summer).to(system=cluster)
-    res = summer_cpu(1, 5)
-    assert res == 6
-
-    torch_summer_cpu = rh.function(fn=summer).to(system=cluster, env=["torch"])
-    torch_res = torch_summer_cpu(1, 5)
-    assert torch_res == 6
-
-
-@pytest.mark.clustertest
-def test_setup_cmds_backwards_compatible(cluster):
-    torch_summer_cpu = rh.function(fn=summer).to(system=cluster, env=["torch"])
-    torch_res = torch_summer_cpu(1, 5)
-    assert torch_res == 6
+    @pytest.mark.level("local")
+    # TODO: currently keep warm does not really keeping the cluster warm, need to think if this method is even
+    #  necessary and whether should we test it
+    def test_keep_warm(self, cluster, caplog):
+        remote_summer = rh.function(summer).to(system=cluster)
+        keep_warm = remote_summer.keep_warm(1)
+        assert keep_warm == remote_summer
+        assert remote_summer(3, 9) == 12
+
+    @unittest.skip("Does not work properly, should be tested after secrets refactor.")
+    @pytest.mark.level("local")
+    def test_send_secrets(self, cluster):
+        load_dotenv()
+        headers = {"Authorization": f"Bearer {os.getenv('TEST_TOKEN')}"}
+        rh.Secrets.put(
+            provider="aws",
+            secret={"access_key": "123", "secret_key": "4567"},
+            headers=headers,
+        )
+        rh.function(fn=summer).to(system=cluster).send_secrets(providers=["aws"])
+        assert True
```

Comment:
Let's cut both of these tests for now. The second one is really problematic because it messes with local secrets.

Reply:
Removed. Also added a TODO comment, as a reminder to consider to test send secrets after Secretes refactor, if it will be necessary.